agent/src/main/java/org/openjdk/jmc/agent/impl/ResolvedConvertable.java

```diff
@@ -0,0 +1,149 @@
+ * the distribution.
+ * 
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be used to
+ * endorse or promote products derived from this software without specific prior written permission.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ * WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.openjdk.jmc.agent.impl;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.Parameter;
+
+import org.objectweb.asm.Type;
+import org.openjdk.jmc.agent.Convertable;
+
+public final class ResolvedConvertable extends AbstractConvertable implements Convertable {
+	public static final String DEFAULT_CONVERTER_METHOD = "convert";
+	private final Class<?> converterClass;
+	private final Method converterMethod;
+
+	public ResolvedConvertable(String converterDefinition, Class<?> typeToConvert) throws MalformedConverterException {
+		super(converterDefinition);
+		if (typeToConvert == null) {
+			throw new MalformedConverterException("Type to convert cannot be null!");
+		}
+		String className = resolveClassName(converterDefinition);
+		Class<?> tmpClass = null;
+		try {
+			if (converterDefinition != null) {
+				tmpClass = Class.forName(className);
+			}
+		} catch (ClassNotFoundException e) {
+			throw new MalformedConverterException("Converter must convert to an existing class!", e);
+		}
+		this.converterClass = tmpClass;
+		this.converterMethod = getConvertMethod(tmpClass, converterDefinition, typeToConvert);
+	}
+
+	public ResolvedConvertable(String converterDefinition, Type type) throws MalformedConverterException {
+		this(converterDefinition, getClassFromType(type));
+	}
+
+	public Class<?> getConverterClass() {
+		return converterClass;
+	}
+
+	public Method getConverterMethod() {
+		return converterMethod;
+	}
+
+	private static Method getConvertMethod(Class<?> converterClass, String converterDefinition, Class<?> originalType)
+			throws MalformedConverterException {
+		String methodName = resolveMethodName(converterDefinition);
+
+		if (converterClass == null) {
+			return null;
+		}
+		for (Method m : converterClass.getDeclaredMethods()) {
+			if (methodName.equals(m.getName())) {
+				if ((!Modifier.isStatic(m.getModifiers())) || m.getParameterCount() != 1) {
```

Comment:
I'd suggest to also filter out abstract (or interface) methods here, with a more meaningful exception as then trying to invoke them later on.